src/parameter_bridge.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright 2017 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <list>
+#include <string>
+
+// include ROS 1
+#ifdef __clang__
+# pragma clang diagnostic push
+# pragma clang diagnostic ignored "-Wunused-parameter"
+#endif
+#include "ros/ros.h"
+#ifdef __clang__
+# pragma clang diagnostic pop
+#endif
+
+// include ROS 2
+#include "rclcpp/rclcpp.hpp"
+
+#include "ros1_bridge/bridge.hpp"
+
+
+int main(int argc, char * argv[])
+{
+  // ROS 1 node
+  ros::init(argc, argv, "ros_bridge");
+  ros::NodeHandle ros1_node;
+
+  // ROS 2 node
+  rclcpp::init(argc, argv);
+  auto ros2_node = rclcpp::node::Node::make_shared("ros_bridge");
+
+  std::list<ros1_bridge::BridgeHandles> all_handles;
+
+  const char * parameter_name = "topics";
+  if (argc > 1) {
+    parameter_name = argv[1];
+  }
+
+  XmlRpc::XmlRpcValue topics;
+  if (
+    ros1_node.getParam(parameter_name, topics) &&
+    topics.getType() == XmlRpc::XmlRpcValue::TypeArray)
+  {
+    for (size_t i = 0; i < static_cast<size_t>(topics.size()); ++i) {
+      std::string topic_name = static_cast<std::string>(topics[i]["topic"]);
```

Comment:
There's a check that the topics param is an array, but not that the elements of the array are dictionaries. this will crash with an XmlRpcException if they're not.